assets/src/components/CustomLink/CustomLink.js

```diff
@@ -0,0 +1,48 @@
+import {Component} from '@wordpress/element';
+import {URLInput} from '@wordpress/block-editor'
+import {TextControl as BaseTextControl} from '@wordpress/components';
+import withCharacterCounter from '../../components/withCharacterCounter/withCharacterCounter';
+
+const TextControl = withCharacterCounter( BaseTextControl );
+
+export class CustomLink extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+
+  isCustomLinkValid() {
+    return !!this.props.value ? this.props.value.substr(0, 8).toLowerCase() === 'https://' : true;
+  }
+
+  render() {
+
+    const {__} = wp.i18n;
+    const allowSearch = !!this.props.allowSearch;
```

Comment:
Why not `const { allowSearch } = this.props;`? You could then also add the other props to that.